flink-ml-lib/src/test/java/org/apache/flink/ml/recommendation/SwingTest.java

```diff
@@ -0,0 +1,232 @@
+    public void testDataType() {
+        List<Row> rows =
+                new ArrayList<>(Arrays.asList(Row.of(0, "10"), Row.of(1, "11"), Row.of(2, "")));
+        DataStream<Row> dataStream = env.fromCollection(rows);
+        Schema schema =
+                Schema.newBuilder()
+                        .column("f0", DataTypes.INT())
+                        .column("f1", DataTypes.BIGINT())
+                        .build();
+        Table data = tEnv.fromDataStream(dataStream, schema).as("user_id", "item_id");
+        Table[] swingResultTables =
+                new Swing()
+                        .setItemCol("item_id")
+                        .setUserCol("user_id")
+                        .setOutputCol("item_score")
+                        .setMinUserItems(1)
+                        .transform(data);
+
+        List<Row> results = IteratorUtils.toList(swingResultTables[0].execute().collect());
+    }
+
+    @Test(expected = RuntimeException.class)
+    public void testNumberFormat() {
+        List<Row> rows =
+                new ArrayList<>(
+                        Arrays.asList(
+                                Row.of(0L, 10L),
+                                Row.of(null, 12L),
+                                Row.of(1L, 13L),
+                                Row.of(3L, 12L)));
+        DataStream<Row> dataStream = env.fromCollection(rows);
+        Schema schema =
+                Schema.newBuilder()
+                        .column("f0", DataTypes.BIGINT())
+                        .column("f1", DataTypes.BIGINT())
+                        .build();
+        Table data = tEnv.fromDataStream(dataStream, schema).as("user_id", "item_id");
+        Swing swing = new Swing().setItemCol("item_id").setUserCol("user_id").setMinUserItems(1);
+        Table[] swingResultTables = swing.transform(data);
+        List<Row> results = IteratorUtils.toList(swingResultTables[0].execute().collect());
+    }
+
+    @Test
+    public void testOutputSchema() {
+        Swing swing =
+                new Swing()
+                        .setItemCol("item_id")
+                        .setUserCol("user_id")
+                        .setOutputCol("item_score")
+                        .setMinUserItems(1);
+        Table[] swingResultTables = swing.transform(trainData);
+        Table output = swingResultTables[0];
+
+        assertEquals(
+                Arrays.asList("item_id", "item_score"),
+                output.getResolvedSchema().getColumnNames());
+    }
+
+    @Test
+    public void testFewerItemCase() {
+        Swing swing = new Swing().setItemCol("item_id").setUserCol("user_id").setMinUserItems(5);
+        Table[] swingResultTables = swing.transform(trainData);
+        Table output = swingResultTables[0];
+        List<Row> results = IteratorUtils.toList(output.execute().collect());
+        Assert.assertEquals(0, results.size());
+    }
+
+    @Test
+    public void testFitAndPredict() throws Exception {
```

Comment:
Swing doesn't fit and predict, it just transforms. Please rename the test function to `testTransform()`.